qa/tasks/cephfs/test_nfs.py

```diff
@@ -334,6 +363,53 @@ def _write_to_read_only_export(self, pseudo_path, port, ip):
         else:
             self.fail('expected write to a read-only export to fail')
 
+    def _create_cluster_with_fs(self, fs_name, mnt_pt=None):
+        """
+        create a cluster along with fs and mount it to the path supplied
+        :param fs_name: name of CephFS volume to be created
+        :param mnt_pt: mount fs to the path
+        """
+        self._test_create_cluster()
+        self._cmd('fs', 'volume', 'create', fs_name)
+        with contextutil.safe_while(sleep=5, tries=30) as proceed:
+            while proceed():
+                output = self._cmd(
+                    'orch', 'ls', '-f', 'json',
+                    '--service-name', f'mds.{fs_name}'
+                )
+                j = json.loads(output)
+                if j[0]['status']['running']:
+                    break
+        if mnt_pt:
+            tries = 3
+            while True:
```

Comment:
Why is this not use the `safe_while`  helper?

Reply:
the code that uses safe_while is a different operation that takes care of making sure the mds service is up while this code is for making sure ceph-fuse is mounted, also i didnt want to give as much sleep time and tries as the above code

Reply:
although i too feel this part can be improvised, I can do that tomorrow

Reply:
@vshankar
`safe_while` would need sleep of 5 secs and 30 tries for making sure mds service is up and running so i cannot bring the wait time down here (to what it is for this code), this code deals with mounting ceph-fuse, i dont thinking giving it 180 secs is worth it, in my runs, max 5-6 secs are needed for mds to show up and accept mount request, even after 9 secs(3 tries, sleep of 3 at each) if it's not able to mount then there's something broken, also if i make this part of the `safe_while` as it is then it would look ugly (while loop inside a while loop) and  it's behaviour will be same as current code because the inner loop would indeed raise when it is out of `tries`.

We can either:
1) make this part of `safe_while` and agree to provide 180 secs wait which would elimiate loop inside loop case
2) have a `safe_while(tries=3, sleep=3)` block outside of above `safe_while` for the this code thus eliminating use of `while()` here
3) keep the code as it is as it's not that bad

Reply:
I would vote for (2) - it might look a bit weird, but you can always move that to a separate routine.

Reply:
done, PTAL